comp/core/autodiscovery/listeners/snmp_test.go

```diff
@@ -34,6 +34,7 @@ func TestSNMPListener(t *testing.T) {
 
 	mockConfig := config.Mock(t)
 	mockConfig.SetWithoutSource("snmp_listener", listenerConfig)
+	mockConfig.SetWithoutSource("network_devices.snmp_listener", listenerConfig)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** 
I think I'd prefer us to use only the latter in these tests to keep each test as small and precise as possible. While having a separate test (or tests!) that asserts that we do the override correctly
‏
```suggestion
	mockConfig.SetWithoutSource("network_devices.snmp_listener", listenerConfig)
```

Reply:
I removed all the duplicated tests in [this commit](https://github.com/DataDog/datadog-agent/pull/24644/commits/111213efc988729971a4d6313d04df13943b0915)